### PR TITLE
Strip illegal XML characters from failure messages

### PR DESCRIPTION
**Pull request by justinbarry**

### CODE REVIEW

File: __tests__/testResultProcessor.test.js [hunk 1]
```diff
@@ -41,4 +45,21 @@ describe('jest-junit', () => {
...
+    const failingTestInvalidXMLCharacter = require('../__mocks__/failing-test-invalid-xml-character.json');
+    testResultProcessor(failingTestInvalidXMLCharacter);
+
+    // Ensure fs.writeFileSync is called
+    expect(fs.writeFileSync).toHaveBeenCalledTimes(1);
+
+    // Ensure file would have been generated
+    expect(fs.writeFileSync).toHaveBeenLastCalledWith(path.resolve('junit.xml'), expect.any(String));
+
+    console.log(fs.writeFileSync.mock.calls[0][1]);
```
Comment (palmerj3) [inline __tests__/testResultProcessor.test.js]: Please remove

File: utils/buildJsonResults.js [hunk 2]
```diff
@@ -4,6 +4,39 @@ const stripAnsi = require('strip-ansi');
 const constants = require('../constants/index');
 const path = require('path');
 
+// Borrowed from here: https://gist.github.com/john-doherty/b9195065884cdbfd2017a4756e6409cc/
```
Comment (palmerj3) [inline utils/buildJsonResults.js]: I would much prefer not doing direct string manipulation. It appears that libxmljs might be able to handle this - could you see if that's possible?

See: https://github.com/libxmljs/libxmljs/blob/master/src/xml_document.cc#L325

File: __tests__/testResultProcessor.test.js [hunk 3]
```diff
@@ -41,4 +45,21 @@ describe('jest-junit', () => {
...
+
+    // Ensure generated file is valid xml
+    const xmlDoc = libxmljs.parseXml(fs.writeFileSync.mock.calls[0][1]);
+    expect(xmlDoc).toBeTruthy();
```
Comment (palmerj3) [inline __tests__/testResultProcessor.test.js]: This test doesn't actually confirm that xml 1.0 invalid characters aren't present

File: __tests__/testResultProcessor.test.js (hunk 1 above)
Comment (justinbarry) [reply __tests__/testResultProcessor.test.js]: Good catch. Done.

File: __tests__/testResultProcessor.test.js (hunk 3 above)
Comment (justinbarry) [reply __tests__/testResultProcessor.test.js]: It doesn't directly confirm but if you remove `removeXMLInvalidChars` from [here](https://github.com/jest-community/jest-junit/pull/94/files#diff-de6bb0568cec60450b473561b52bbf0dR183) the xml file will fail to validate.

File: utils/buildJsonResults.js (hunk 2 above)
Comment (justinbarry) [reply utils/buildJsonResults.js]: The encoding lib is probably where this should go. 

Yeah, I was arguing with myself on where this should go. Now that you mention is, it should live in https://github.com/dylang/node-xml. No need for this library to know which characters are valid and which aren't. I'll close this PR.

**Comment by justinbarry** [comment]
This should probably go into https://github.com/dylang/node-xml. I'll try that.

**Comment by justinbarry** [comment]
@palmerj3 Thank you for looking at this PR! 

This weird edge case that probably doesn't affect a significant portion of this modules user base.

**Pull request closed without merging**